src/contexts/walletconnect.tsx

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2018 - 2022 The Alephium Authors
+This file is part of the alephium project.
+
+The library is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+The library is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with the library. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+import WalletConnectClient, { CLIENT_EVENTS } from '@walletconnect/client'
+import { SessionTypes } from '@walletconnect/types'
+import { createContext, Dispatch, FC, SetStateAction, useContext, useEffect, useState } from 'react'
+
+import { useAddressesContext } from '../contexts/addresses'
+import { SendTransactionData } from '../modals/SendModal'
+import { useGlobalContext } from './global'
+
+export interface ContextType {
+  isWalletConnectModalOpen: boolean
+  setIsWalletConnectModalOpen: Dispatch<SetStateAction<boolean>>
+  walletConnect?: WalletConnectClient
+  setWalletConnect: Dispatch<SetStateAction<WalletConnectClient | undefined>>
+  dappTransactionData?: SendTransactionData
+  requestEvent?: SessionTypes.RequestEvent
+}
+
+export const initialContext: ContextType = {
+  isWalletConnectModalOpen: false,
+  setIsWalletConnectModalOpen: () => false,
+  walletConnect: undefined,
+  setWalletConnect: () => undefined,
+  dappTransactionData: undefined,
+  requestEvent: undefined
+}
+
+export const Context = createContext<ContextType>(initialContext)
+
+export const WalletConnectContextProvider: FC = ({ children }) => {
+  const { setIsSendModalOpen, settings } = useGlobalContext()
+  const { addresses } = useAddressesContext()
+  const [isWalletConnectModalOpen, setIsWalletConnectModalOpen] = useState(false)
+  const [walletConnect, setWalletConnect] = useState<WalletConnectClient | undefined>(undefined)
+  const [dappTransactionData, setDappTransactionData] = useState<SendTransactionData | undefined>(undefined)
+  const [requestEvent, setRequestEvent] = useState<SessionTypes.RequestEvent | undefined>(undefined)
+
+  useEffect(() => {
+    if (walletConnect === undefined) {
+      WalletConnectClient.init({
+        controller: true,
+
+        // TODO: configurable during building?
+        projectId: '6e2562e43678dd68a9070a62b6d52207',
```

Comment:
Need to make this configurable

Reply:
It comes from WalletConnect - you have to get one from them, I guess for their relay to work properly and to avoid spam on their network...

Reply:
Added a better todo, but it should be done later and not important now.